sdk/core/azure-core/src/main/java/com/azure/core/util/polling/SimpleSyncPoller.java

```diff
@@ -0,0 +1,284 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util.polling;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.polling.implementation.PollContextRequiredException;
+import reactor.core.Exceptions;
+
+import java.time.Duration;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+/**
+ *
+ * Simple implementation of {@link SyncPoller}.
+ *
+ * @param <T> The type of poll response value
+ * @param <U> The type of the final result of the long running operation
+ */
+public final class SimpleSyncPoller<T, U> implements SyncPoller<T, U> {
```

Comment:
Do we need a separate type for this? Can we add a constructor overload to [DefaultSyncPoller](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core/src/main/java/com/azure/core/util/polling/DefaultSyncPoller.java) to support the sync poll, cancel and fetch result operations?

The interface methods can then check whether the sync stack should be used or async stack should be used and call appropriate private methods. Another public type called `SimpleSyncPoller` may be confusing and may not be considered the default (because there's a type called DefaultSyncPoller).